src/main/java/Dude/Dude.java

```diff
@@ -0,0 +1,601 @@
+package Dude;
```

Comment:
I like that there is a minimal number of nested loops, which shows your good coding ability.